pkg/resources/kafka/configmap.go

```diff
@@ -356,6 +394,9 @@ func (r Reconciler) generateBrokerConfig(id int32, brokerConfig *v1beta1.BrokerC
 
 	// Merge operator generated configuration to the final one
 	if opGenConf != nil {
+		// When there is custom super.users configuration we merge its value into the Koperator generated one
+		// thus finalBrokerConfig contains merged super.users value.
+		mergeSuperUsersPropertyValue(finalBrokerConfig, opGenConf)
```

Comment:
**Required**: We **MUST** either update the comment here or update the function logic because with the current implementation the `target` contains the final value (line 383) which among the arguments here is `opGenConf`, not `finalBrokerConfig`, but the last line of the comment makes it sound it should be `finalBrokerConfig` that contains the merged super.users value.
(Whichever the following logic builds upon should be the consistent truths in both logic and comment.)

Reply:
Oh I see the next line we merge the other way, that spins it a little.
I think we should try to find a better phrasing still, but I understand where it went wrong here first.